drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -1270,6 +1272,35 @@ static void usbd_work_handler(struct k_work *item)
 			}
 			break;
 		case USBD_EVT_SOF:
+			if (nrfx_usbd_ep_enable_check(NRFX_USBD_EPOUT8)) {
```

Comment:
Probably you could move ISO related code to own function e.g. `usbd_sof_trigger_iso_read()`.